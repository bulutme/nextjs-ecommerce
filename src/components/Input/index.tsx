import React, { InputHTMLAttributes } from "react";
import styled from "styled-components";

interface Props extends InputHTMLAttributes<HTMLInputElement> {}

const StyledInput = styled.input`
  flex: 1;
  padding: 0.8rem;
  font-size: 1rem;
  border: 1px solid #ccc;
  border-radius: 0.375rem;
  outline: none;
  height: 100%;

  &:focus {
    border-color: #3bb77e;
    box-shadow: 0 0 0 3px rgba(52, 152, 219, 0.2);
  }
`;

const Input = (props: Props) => {
  return <StyledInput {...props} />;
};

export default Input;
