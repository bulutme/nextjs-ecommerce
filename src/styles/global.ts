"use client";

import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
  }

  html, body, #__next {
    height: 100%;
  }

  a {
    text-decoration: none;
    width: fit-content;
  }

  ul {
    list-style-type: none
  }
`;

export default GlobalStyles;
