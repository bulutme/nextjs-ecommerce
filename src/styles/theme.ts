export const theme = {
  colors: {
    black: "#000",
    white: "#fff",
    gray: "#808080",
    grayLight: "#d3d3d3",
    border: "#e2e8f0",
    primary: "#3bb77e",
    primaryDark: "#2b8a5e",
    error: "#dc2626",
    info: "#0891b2",
    success: "#16a34a",
  },
  screens: {
    "2xs": "400px",
    xs: "480px",
    sm: "576px",
    md: "768px",
    lg: "992px",
    xl: "1200px",
    "2xl": "1440px",
    "3xl": "1560px",
  },
  fontWeight: {
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
  },
};
