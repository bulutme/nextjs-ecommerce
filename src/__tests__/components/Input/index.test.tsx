import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@/styles/theme";
import Input from "@/components/Input";

const renderWithTheme = (component: React.ReactElement) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

describe("Input Component", () => {
  test("renders correctly", () => {
    const { container } = renderWithTheme(<Input />);
    expect(container).toMatchSnapshot();
  });

  test("applies focus styles correctly", () => {
    const { getByTestId } = renderWithTheme(<Input data-testid="input" />);
    const input = getByTestId("input");

    expect(input).not.toHaveStyle(`
      border-color: ${theme.colors.primary};
      box-shadow: 0 0 0 3px rgba(52, 152, 219, 0.2);
    `);

    input.focus();

    expect(input).toHaveStyle(`
      border-color: ${theme.colors.primary};
      box-shadow: 0 0 0 3px rgba(52, 152, 219, 0.2);
    `);
  });
});
