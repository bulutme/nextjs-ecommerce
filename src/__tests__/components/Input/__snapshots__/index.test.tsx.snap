// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders correctly 1`] = `
<div>
  <input
    class="sc-acef6806-0 bmACrq"
  />
</div>
`;
